cel/program.go

```diff
@@ -203,8 +203,9 @@ func (gen *progGen) Eval(vars interpreter.Activation) (ref.Val, EvalDetails, err
 
 	// Evaluate the input, returning the result and the 'state' as EvalDetails.
 	v, _, err := p.Eval(vars)
+	det := &evalDetails{state: state}
```

Comment:
Shouldn't this be returned on line 201 as well?

Reply:
Good question. The first block of code configures the program execution environment, but didn't do evaluation. The second block actually runs the expression eval

Reply:
I've updated the code to always return `EvalDetails` to improve the consistency in behavior. I've also added a factory construction test within the `Program()` call to reduce the likelihood that an error in `Interpretable` construction is deferred until the first `Eval()` call. Thanks for the feedback!